pkg/action/sysext.go

```diff
@@ -0,0 +1,384 @@
+package action
+
+import (
+	"fmt"
+	"github.com/distribution/reference"
+	"github.com/kairos-io/kairos-agent/v2/pkg/config"
+	"github.com/kairos-io/kairos-sdk/types"
+	"github.com/twpayne/go-vfs/v5"
+	"net/url"
+	"os"
+	"path/filepath"
+	"regexp"
+)
+
+// Implementation details for not trusted boot
+// sysext are stored under
+// /var/lib/kairos/extensions/
+// we link them to /var/lib/kairos/extensions/{active,passive} depending on where we want it to be enabled
+// Immucore on boot after mounting the persistent dir, will check those dirs\
+// it will then create the proper links to them under /run/extensions
+// This means they are enabled on boot and they are ephemeral, nothing is left behind in the actual sysext dirs
+// This prevents us from having to clean up in different dirs, we can just do cleaning in our dirs (remove links)
+// and on reboot they will not be enabled on boot
+// So all the actions (list, upgrade, download, remove) will be done on the persistent dir
+// And on boot we dinamycally link and enable them based on the boot type (active,passive) via immucore
+
+// Trusted boot works very differently
+// In there, we copy the sysext into the EFI dir under the entry /EFI/kairos/$EFI_FILE.efi.extra.d/EXTENSION_FILE
+// That makes it so systemd-boot will see it, measure it and pass it to the initramfs under the /.extra/sysext dir
+// That means two things
+// We cannot soft link it as EFI partition is FAT
+// To add/remove/update we need to deal with the actual files
+// Currently we are NOT using the measuring of the sysextensions so maybe we could move into a unified system with the same behaviour as above?
+// The sysextensions in Trusted boot are SIGNED so the measurement is not that important and immucore will refuse to move
+// them into /run/extensions if the signature doesnt match, so the measurements are not that important currently and
+// maybe they are not useful if sysext is the way of upgrading your system
+
+// So potentially we have a unified system where we have the same behaviour for both, but that measn that
+// anything provided by sysext will NOT be available until the initramfs stage where we enable the sysext service
+// we can potentially move the sysext service as early as possible once we mount the persistent dir
```

Comment:
I think we should just decide and update the comments to reflect the final decision. If measurement doesn't buy us anything, let's unify the 2 "modes". Since the directory where systemd expects the extensions in order to load them is always the same (trusted boot or not), I guess the reason to use symlinks in the non-trusted boot mode is to avoid copying files (for speed). Is that correct?

Reply:
yes, this comments should go away it was thefirst part only when I was starting